### PR TITLE
Added All Tall, All Small and All or Nothing at All bets

### PR DESCRIPTION
**Pull request by amortization**
I added 3 new bets, AllTall, AllSmall, and AllOrNothingAtAll. We don't have these bets locally and I haven't played them myself, but I saw that they are very popular in Vegas from the [Reddit/Twitter Craps Minimums spreadsheet](https://docs.google.com/spreadsheets/d/1txvaruxsoprcfgHOXkNh4MSqwxz3GYIciX_8TNOigGk/edit#gid=0). I took most of the information regarding how these bets work and the rules for them from [here](https://www.little-creek.com/casino/bonus-craps.php#:~:text=More%20Wagering%2C%20More%20Winning&text=Played%20just%20like%20a%20regular,a%20whopping%20175%20to%201.).

For anyone more familiar with this bet, do the odds listed change from the ones listed (34 and 175) or are those pretty consistent across tables? If not, we should probably add a way to edit the payouts for those, perhaps utilizing the table.payouts dict. Also, are there any requirements for when the bet is place (ex. requiring a new shooter or that the point is on/off)? 

If neither of those are required these have tests and should be ready to go.

### CODE REVIEW

**Comment by skent259** [comment]
Based on https://wizardofodds.com/games/craps/side-bets/bonus-craps/ and https://www.crapspit.org/small-tall-all-bets/, I think that there are two payouts that are common:

- 34 to 1, 174 to 1
- 30 for 1, 150 for 1 (i.e. 29:1 and 149:1)

Of the two, I'm not sure which is more common but I'm checking on that. It would be ideal to add the option to `table.payouts` and the bet.  

The bet can only be placed when there is a new shooter (as far as I am aware), so it needs similar logic to the Fire bet in PR #12

**Comment by skent259** [comment]
It seems the 30/30/150 payouts are more common, so let's start with those and improve the payout options later with issue #13.

File: crapssim/bet.py [hunk 1]
```diff
@@ -550,3 +550,59 @@ def _update_bet(self, table_object: "Table", dice_object: Dice) -> tuple[str | N
...
+        return table.new_shooter
+
+
+class AllTall(AllBet):
+    def __init__(self, bet_amount):
+        super().__init__(bet_amount)
+        self.name = 'AllTall'
+        self.bet_amount = bet_amount
+        self.numbers = [8, 9, 10, 11, 12]
+        self.payoutratio = 30
```
Comment (skent259) [inline crapssim/bet.py:590]: I should have clarified, but this should be `self.payoutratio = 29`, with 29 for AllSmall and 149 for AllOrNothingAtAll.  This reflects the 30/30/150 for 1, i.e. 29/29/149 to 1.

File: crapssim/bet.py [hunk 2]
```diff
@@ -550,3 +550,59 @@ def _update_bet(self, table_object: "Table", dice_object: Dice) -> tuple[str | N
 
     def allowed(self, table: 'Table') -> bool:
         return table.new_shooter
+
+
+class AllBet(Bet):
```
Comment (skent259) [inline crapssim/bet.py:555]: Maybe a bit pedantic, but what about changing this naming to `class ATS(Bet)`, and then changing the sub bets to `Tall(ATS)`, `Small(ATS)`, and `All(ATS)`?  I think this better reflects the naming conventions I see and leads to an easy to read strategy such as:

```
player.bet(All(player.unit))
player.bet(Tall(player.unit))
player.bet(Small(player.unit))
```

**Review by skent259 — CHANGES_REQUESTED** [review]
A few minor comments, otherwise this PR should be good to go!  Definitely a great addition to have

File: crapssim/bet.py (hunk 2 above)
Comment (amortization) [reply crapssim/bet.py:555]: Yeah, so I've never actually made this bet so I just copied what was printed on the felt in a google image search for it. I think if that you generally say All, Tall, and Small when making the bet that is consistent with the other bets being what is said when the bet is made (ex: Yo or Boxcars.) 

I'm a little concerned that users won't know the difference between All and ATS when importing and will think that ATS is meant to be All (although currently it's not much better.) Maybe the current AllBet should be something like ATSBase or something like that to differentiate it.

Comment (skent259) [reply crapssim/bet.py:555]: Fair point about ATS as the base class.  I like ATSBase or ATSClass

**Comment by skent259** [comment]
Closing in favor of #37 because new bet changes required many changes

**Comment by skent259** [comment]
Closing in favor of #37  because new bet changes required many changes

**Pull request closed without merging**